app/controllers/application_controller.rb

```diff
@@ -47,10 +47,33 @@ def create_user_event(event_type, user = current_user)
   private
 
   def redirect_on_timeout
-    request_params = request.query_parameters
-    return unless request_params[:timeout]
-    flash[:timeout] = t('notices.session_cleared')
-    redirect_to url_for(request_params.except(:timeout))
+    params = request.query_parameters
+    return unless params[:timeout]
+
+    params[:issuer].present? ? redirect_with_sp : redirect_without_sp
+  end
+
+  def sp_metadata
+    ServiceProvider.from_issuer(request.query_parameters[:issuer]).metadata
+  end
+
+  def sp_name
+    sp_metadata[:friendly_name] || sp_metadata[:agency]
+  end
+
+  def redirect_with_sp # rubocop:disable Metrics/AbcSize
```

Comment:
I personally am happy to disable rubocop everywhere, but I know that @monfresh and @pkarman sometimes try to appease it rather than disable. Any strong opinions here?